src/components/TodoItem.tsx

```diff
@@ -0,0 +1,126 @@
+import React,
+{
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { TodosContext } from '../TodoContext/TodoContext';
+import { Todo } from '../types/Todo';
+
+interface TodoItems {
+  todo: Todo;
+}
+
+export const TodoItem: React.FC<TodoItems> = ({ todo }) => {
+  const {
+    title,
+    completed,
+    id,
+  } = todo;
+
+  const {
+    setCompleted,
+    deleteTodo,
+    saveEditingTitle,
+  } = useContext(TodosContext);
+
+  const [isEdeting, setIsEditing] = useState(false);
+  const [editingTitle, setEditingTitle] = useState(todo.title);
+
+  const titleFocus = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleFocus.current) {
+      titleFocus.current.focus();
+    }
+  }, [isEdeting]);
+
+  const handleOnKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      if (editingTitle.trim()) {
+        saveEditingTitle(todo.id, editingTitle.trim());
+        setEditingTitle(editingTitle.trim());
+        setIsEditing(false);
+      } else {
+        deleteTodo(todo.id);
+      }
+    }
+
+    if (event.key === 'Escape') {
+      if (!editingTitle.trim()) {
+        setIsEditing(false);
+        setEditingTitle(todo.title);
+      } else {
+        setIsEditing(false);
+        setEditingTitle(todo.title);
+      }
+    }
```

Comment:
You do the same things)
```suggestion
      setIsEditing(false);
      setEditingTitle(todo.title)
    }
```